### PR TITLE
{bio}[dummy/] Gctf v1.06

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/38c55b0923eb47bc329d4dc96e98fb94 for a full test report.

File: easybuild/easyconfigs/g/Gctf/Gctf-1.06.eb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+homepage = 'https://www.mrc-lmb.cam.ac.uk/kzhang/'
+description = """Gctf: real-time CTF determination and correction, Kai Zhang, 2016"""
+
+toolchain = {'name': 'dummy', 'version': ''}
+toolchainopts = {'openmp': True}
+
+source_urls = ['http://www.mrc-lmb.cam.ac.uk/kzhang/Gctf/']
+sources = [
+    '%(name)s_v%(version)s_and_examples.tar.gz',
```
Comment (verdurin) [inline easybuild/easyconfigs/g/Gctf/Gctf-1.06.eb]: @akesandgren Please add the missing checksum.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/7bef522a628a8172da72e1e6daaa3b14 for a full test report.

File: easybuild/easyconfigs/g/Gctf/Gctf-wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Wrap around Gctf to load the required toolchain so it can use the
+# expected version of CUDA.
+
+# Save the path to the install tree
+gctf_base=$EBROOTGCTF
+
+module purge > /dev/null 2>&1
```
Comment (boegel) [inline easybuild/easyconfigs/g/Gctf/Gctf-wrapper.sh]: @akesandgren Why not only unload `CUDA`, should be safer than `purge`?

Comment (akesandgren) [reply easybuild/easyconfigs/g/Gctf/Gctf-wrapper.sh]: For Gctf it makes sense, I just noticed that it depends on libstdc++6 so it needs a GCCcore too to be safe against very minimal OS:es.
(On the other hand libstdc++.so.6 is a vital part of all Linux distros that are still used)

File: easybuild/easyconfigs/g/Gctf/Gctf-wrapper.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Wrap around Gctf to load the required toolchain so it can use the
+# expected version of CUDA.
```
Comment (boegel) [inline easybuild/easyconfigs/g/Gctf/Gctf-wrapper.sh]: @akesandgren Comment should be fixed, we're only loading the required CUDA, not any toolchain?

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in this PR)
rescomp2 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/58a080ac7ba3ffa07845b7fea18f2eb2 for a full test report.

File: easybuild/easyconfigs/g/Gctf/Gctf-1.06.eb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+]
+patches = [('Gctf-wrapper.sh', '.')]
+checksums = [
+    'bfb6305f36571f895402bb22a2e8c6c5c4f13e28d3354ff8fe101ec25b351d77',  # Gctf_v1.06_and_examples.tar.gz
+    'bd019bd752ac56b89f9f09ce8e9aab41977be1078145b7afa5faf48f4ef4781b',  # Gctf-wrapper.sh
```
Comment (verdurin) [inline easybuild/easyconfigs/g/Gctf/Gctf-1.06.eb]: This is failing for me:
```
== 2019-05-14 08:46:10,294 filetools.py:724 DEBUG Using blocksize 16777216 for calculating the checksum
== 2019-05-14 08:46:10,294 filetools.py:772 DEBUG Computed sha256 checksum for /tmp/eb-p8hqwK/files_pr8045/g/Gctf/Gctf-wrapper.sh: 62cc5b011476a5005360003c16bcc408b09f18e368d261b95522715d6bc47617 (correct checksum: bd019bd752ac56b89f9f09ce8e9aab41977be1078145b7afa5faf48f4ef4781b)
== 2019-05-14 08:46:10,320 build_log.py:163 ERROR EasyBuild crashed with an error (at ?:124 in __init__): Checksum verification for /tmp/eb-p8hqwK/files_pr8045/g/Gctf/Gctf-wrapper.sh using bd019bd752ac56b89f9f09ce8e9aab41977be1078145b7afa5faf48f4ef4781b failed. (at easybuild/framework/easyblock.py:1777 in checksum_step)
== 2019-05-14 08:46:10,320 easyblock.py:3010 WARNING build failed (first 300 chars): Checksum verification for /tmp/eb-p8hqwK/files_pr8045/g/Gctf/Gctf-wrapper.sh using bd019bd752ac56b89f9f09ce8e9aab41977be1078145b7afa5faf48f4ef4781b failed.
== 2019-05-14 08:46:10,320 easyblock.py:293 INFO Closing log for application name Gctf version 1.06
```

Comment (akesandgren) [reply easybuild/easyconfigs/g/Gctf/Gctf-1.06.eb]: Crap, forgot to update checksum after the change in the comment...

Comment (akesandgren) [reply easybuild/easyconfigs/g/Gctf/Gctf-1.06.eb]: Done

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
rescomp2 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4c958f7ceed51edc462b6646a61a4ed7 for a full test report.

**Comment by verdurin** [comment]
@boegel are you happy to merge this one now?

**Comment by boegel** [comment]
It's better than how it was before, but this approach still gives me the creeps... I can't really think of a real alternative though (other than not using `dummy`).

I do understand the motivation for not using a 'proper' toolchain (since there's nothing to build).

So, let's give some more feedback on this: @vanzod?

**Comment by boegel** [comment]
replaced with #9097

**Pull request closed without merging**